source/docs/software/networktables/stale-data.rst

```diff
@@ -0,0 +1,8 @@
+Stale Data
+==========
+
+NetworkTables acts as a distributed table of name and value pairs. If a name/value pair is added to either the server (robot) or any client (dashboard, co-processor, limelight, etc) it is replicated to all devices. If a name/value pair is deleted from, say, the robot but any NetworkTables clients are still running, then when the robot is restarted, the old values will still appear in the client because they never stopped running and continue to have those values in their tables. When the robot restarts, those old values will be replicated to the robot.
```

Comment:
This is more nuanced than stated.  NT4 has different categories of topic value preservation: persistent, retained, and transient.  Only persistent and retained topic values are preserved by the server when all publishers are gone, and only persistent topic values are preserved through a server reboot. As NT4 defaults to transient publishes, and a client disconnect acts as an “unpublish all”, stale values are rare unless a client specifically marks a topic as retained.

NT3 clients mark all their published topics as retained, so there’s a greater likelihood those values will be stale.  But this still only applies to topics the NT3 client actually writes to.

Reply:
It’s probably also worth noting here that the timestamp column in OutlineViewer can be used to see what topics are being regularly updated.

Reply:
I agree it is more nuanced than written but I don't think its much.

So with Limelight for example (presumably still on NT3? will never know till (if?) we get a new image) would leave all of its stale values?  I am just thinking of one of the most common use cases for teams.

Reply:
However, Limelight isn't changing the data that it sends often. like changing robot programs